src/build.d

```diff
@@ -1002,11 +1002,11 @@ void parseEnvironment()
     auto generated = env["GENERATED"] = dmdRepo.buildPath("generated");
     auto g = env["G"] = generated.buildPath(os, build, model);
     mkdirRecurse(g);
-    env.getDefault("TOOLS_DIR", dmdRepo.dirName.buildPath("tools"));
+    env.setDefault("TOOLS_DIR", dmdRepo.dirName.buildPath("tools"));
 
-    if (env.get("HOST_DMD", null).length == 0)
+    if (env.getDefault("HOST_DMD", null).length == 0)
```

Comment:
`env.get` doesn't read from the environment, only `getDefault` does.

Reply:
Well. that triggers a range error because `HOST_DMD` might be read from the `environment` and is then never set in `env`:

```
core.exception.RangeError@.\src\build.d(1035): Range violation
----------------
0x00007FF732457CAC in onRangeError
0x00007FF7324583BC in d_arraybounds
0x00007FF7323C57EB in build.parseEnvironment at E:\repos\Dlang\forks\dmd\src\build.d(1039)
0x00007FF7323C295E in build.runMain at E:\repos\Dlang\forks\dmd\src\build.d(146)
```

Reply:
Possible workaround:
```patch
diff --git a/src/build.d b/src/build.d
index 0452cfb09..859f9e83d 100755
--- a/src/build.d
+++ b/src/build.d
@@ -1009,6 +1009,8 @@ void parseEnvironment()
         const hostDmd = env.getDefault("HOST_DC", null);
         env["HOST_DMD"] = hostDmd.length ? hostDmd : "dmd";
     }
+    else
+        env["HOST_DMD"] =env.getDefault("HOST_DMD", null);

     // Auto-bootstrapping of a specific host compiler
     if (env.getNumberedBool("AUTO_BOOTSTRAP"))
```